src/tokenMath.ts

```diff
@@ -0,0 +1,47 @@
+/** Converts a token amount to a currency amount
+ * @param args.tokenAmount The token amount
+ * @param args.tokenDecimals The token decimals
+ * @param args.price The price of the token
+ * @param args.priceDecimals The decimal places to use for the price, defaults to 8, any decimals beyond this will be truncated
+ * @returns The currency amount
+ * */
+export function convertTokenToFiat(args: {
+  tokenAmount: bigint;
+  tokenDecimals: number;
+  price: number;
```

Comment:
If `price` is the price of the token, maybe worth naming it `tokenPrice` so it's also immediately clear at call sites just by looking at the invocation?